compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -130,8 +131,6 @@ pub struct Elaborator<'context> {
     /// to the corresponding trait impl ID.
     current_trait_impl: Option<TraitImplId>,
 
-    trait_id: Option<TraitId>,
-
```

Comment:
`Elaborator::trait_id` was set only while resolving trait impls while `FuncMeta::trait_id` is set only for trait functions. Both are used for resolving trait methods either to a specific impl or to the trait itself - so they were incompatible. I found that we already had `Elaborator::current_trait_impl` for trait impls, and the `TraitId` could be retrieved indirectly through that, so this field was redundant and removing it fixed the conflict.